numbat/src/parser.rs

```diff
@@ -2620,6 +2676,58 @@ mod tests {
             },
         );
 
+        parse_as(
+            &["fn double_kef(x) = y where y = x * 2"],
+            Statement::DefineFunction {
+                function_name_span: Span::dummy(),
+                function_name: "double_kef".into(),
+                type_parameters: vec![],
+                parameters: vec![(Span::dummy(), "x".into(), None)],
+                body: Some(identifier!("y")),
+                local_variables: vec![DefineVariable {
+                    identifier_span: Span::dummy(),
+                    identifier: String::from("y"),
+                    expr: binop!(identifier!("x"), Mul, scalar!(2.0)),
+                    type_annotation: None,
+                    decorators: vec![],
+                }],
+                return_type_annotation: None,
+                decorators: vec![],
+            },
+        );
+
+        parse_as(
+            &["fn kefirausaure(x) = z + y
```

Comment:
? :smile:

Reply:
Kefir is just the tiny puppy dog I adopted earlier this year. He now lives in all my tests 
![image](https://github.com/user-attachments/assets/97e936c3-4e39-4583-84ac-bed1e97b7804)
![image](https://github.com/user-attachments/assets/6a392929-f856-471b-9807-7ef958c3d55e)
![image](https://github.com/user-attachments/assets/64781723-4d6e-4ac2-aee6-0119397d1aaf)